### PR TITLE
scripts: convert helper scripts to python3

### PR DESCRIPTION
**Pull request by punitvara**
This patch ports helper scripts from python2 to python3
with following changes:

- print should be used with () in python3
- resolved bytes-like object is required, not 'str'
- added python3 header

Signed-off-by: punit vara <punit.vara@intel.com>

### CODE REVIEW

File: scripts/checkconfig.py [hunk 1]
```diff
@@ -62,7 +62,7 @@
 
 def separate_location_lines(dirs):
     for dir in dirs:
-        print "    ", dir
+        print ("    ", dir)
```
Comment (pfalcon) [inline scripts/checkconfig.py]: Please remove space between "print" and "(", print is a normal function in Python3, and should be used as such (see e.g. 2 lines above, `separate_location_lines(dirs)`)

File: scripts/checkconfig.py [hunk 2]
```diff
@@ -100,7 +100,7 @@ def search_config_in_file(tree, items, completelog, exclude):
             if (fname.endswith(".c") or
             fname.endswith(".h") or
             fname.endswith(".S")):
-                with open(os.path.join(dirName, fname), "r") as f:
+                with open(os.path.join(dirName, fname), encoding="latin-1") as f:
```
Comment (pfalcon) [inline scripts/checkconfig.py]: May I suggest just using "utf-8" encoding as the "default" one?

Comment (pfalcon) [inline scripts/checkconfig.py]: `'utf8'` then as used in changes below

**Review by pfalcon — CHANGES_REQUESTED** [review]
Some codestyle fixes are needed to comply with PEP8 (no need for space between function name (`print`) and open paren).

File: scripts/checkconfig.py (hunk 1 above)
Comment (punitvara) [inline scripts/checkconfig.py]: Done. Is there anyway to check coding style errors in python scripts like we have checkpatch.pl for c files ?

File: scripts/checkconfig.py [hunk 3]
```diff
@@ -84,7 +84,7 @@ def search_kconfig_items(items, name, completelog):
 
         elif item.is_comment():
             if (item.get_text() == name):
-                print completelog
+                print (completelog)
```
Comment (punitvara) [inline scripts/checkconfig.py]: Done

File: scripts/checkconfig.py (hunk 2 above)
Comment (punitvara) [inline scripts/checkconfig.py]: Done

**Review by nashif — APPROVED** [review]

**Pull request merged**